packages/core-contracts/src/contracts/FleetDepositManager.sol

```diff
@@ -0,0 +1,297 @@
+    //////////////////////////////////////////////////////////////*/
+
+    /// @notice The BridgeRouter that manages bridge adapters
+    IBridgeRouter public bridgeRouter;
+
+    /// @notice HarborCommand contract address for fleet commander validation
+    address public immutable harborCommand;
+
+    /// @notice Fleet deposit message type identifier
+    bytes32 public constant FLEET_DEPOSIT_TYPE = keccak256("FLEET_DEPOSIT");
+
+    /*//////////////////////////////////////////////////////////////
+                              CONSTRUCTOR
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * @notice Initializes the FleetDepositManager
+     * @param _bridgeRouter Address of the BridgeRouter
+     * @param _accessManager Address of the ProtocolAccessManager
+     * @param _harborCommand Address of the HarborCommand contract for fleet commander validation
+     */
+    constructor(
+        address _bridgeRouter,
+        address _accessManager,
+        address _harborCommand
+    ) ProtocolAccessManaged(_accessManager) {
+        if (_bridgeRouter == address(0)) revert InvalidParams();
+        if (_harborCommand == address(0)) revert InvalidParams();
+        bridgeRouter = IBridgeRouter(_bridgeRouter);
+        harborCommand = _harborCommand;
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                         GOVERNANCE FUNCTIONS
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * @notice Updates the bridge router address
+     * @param _newBridgeRouter Address of the new BridgeRouter
+     * @dev Only callable by governance
+     */
+    function setBridgeRouter(address _newBridgeRouter) external onlyGovernor {
+        if (_newBridgeRouter == address(0)) revert InvalidParams();
+
+        address oldRouter = address(bridgeRouter);
+        bridgeRouter = IBridgeRouter(_newBridgeRouter);
+
+        emit BridgeRouterUpdated(oldRouter, _newBridgeRouter);
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                            CORE FUNCTIONS
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * @notice Initiates a deposit to a fleet on a target chain using the specified bridge adapter
+     * @param bridgeAdapter Address of the bridge adapter to use (user's choice)
+     * @param destinationChainId Target chain ID where the FleetCommander is deployed
+     * @param asset Asset to bridge and deposit
+     * @param amount Amount to bridge and deposit
+     * @param fleetCommander FleetCommander address on destination chain
+     * @param shareRecipient Address to receive the FleetCommander shares
+     * @param referralCode Optional referral code for tracking
+     * @param adapterParams Bridge-specific adapter parameters
+     * @return operationId Unique identifier for this cross-chain deposit operation
+     */
+    function initiateDepositToTargetChainFleet(
```

Comment:
why is the corsschain deposit using adapter directly if the `bridgeRouter` is the main entry point for the cross chain ecosystem ?

Reply:
Because the bridgeRouter is currently locked down for use by the keeper via the queue.

So, we can either go direct a supporting adapter... or extend bridge router with a permissionless execute method that skips queue. For speed, I opted for direct to adapter.